openpype/hosts/nuke/api/plugin.py

```diff
@@ -98,10 +93,13 @@ def check_existing_subset(self, subset_name, instance_data):
         """Check if existing subset name already exists."""
         exists = False
         for node in nuke.allNodes(recurseGroups=True):
+            # make sure testing node is having instance knob
+            if INSTANCE_DATA_KNOB not in node.knobs().keys():
+                continue
             node_data = get_node_data(node, INSTANCE_DATA_KNOB)
+            # test if subset name is matching
```

Comment:
What happens if the current subset name is `renderMain` but the one on the node in `renderMain01`. Are we sure we want to do a `subset_name in `node_data.get("subset")` check.

Additionally, `subset_name` in `None` will error, so the `.get("subset")` will still error.

Did we mean to do `if subset_name == node.get("subset")` ?

Reply:
> Did we mean to do if subset_name == node.get("subset") ?

Yeah good point, I will change this condition, with the exception that we need to do it on `node_data` rather on node name. It might happen that user will duplicate a node with name `renderMain` with node data `renderMain` and create second node with node name `renderMain01` but the node data will still remain on subset `renderMain`.  I see that more usefull might be to drive subset name from node name but since we store and collect the name from node data we should be checking it from there. We might change that in future perhpas but it will be done with different [issue ](https://github.com/ynput/OpenPype/issues/4329) and PR.